pkg/publish/options.go

```diff
@@ -120,3 +122,24 @@ func Insecure(b bool) Option {
 		return nil
 	}
 }
+
+// WithExplicitRepoHost is a functional option for pinning the repository to a specific
+// host, for example if the repository is talked to through a local tunnel.
+func WithExplicitRepoHost(host string) Option {
```

Comment:
This feels like a very niche option -- can you use `WithTransport` to do this instead?

Reply:
Done.